vendor/k8s.io/client-go/tools/clientcmd/client_config.go

```diff
@@ -608,19 +608,24 @@ func (config *inClusterClientConfig) Possible() bool {
 		err == nil && !fi.IsDir()
 }
 
-// BuildConfigFromFlags is a helper function that builds configs from a master
-// url or a kubeconfig filepath. These are passed in as command line flags for cluster
-// components. Warnings should reflect this usage. If neither masterUrl or kubeconfigPath
-// are passed in we fallback to inClusterConfig. If inClusterConfig fails, we fallback
-// to the default config.
+// InClusterBuildConfig is a helper function that builds configs by trying the InClusterConfig().
```

Comment:
You shouldn't change the vendored code, because next time we update to a new version of k8s client libraries these changes will be lost / overwritten. If you'd like to add a helper function it should be somewhere underneath the [`pkg`](https://github.com/googleforgames/agones/tree/main/pkg) directory.

Reply:
I've added a new package called `k8sconfig` inside `pkg/util` and `make lint` is indicating naming convention issue.

```
pkg/util/k8sconfig/k8s_client_config.go:26:27: var-naming: func parameter masterUrl should be masterURL (revive)
func InClusterBuildConfig(masterUrl, kubeconfigPath string) (*restclient.Config, error) {
```

Am I in the right direction?

Reply:
Yep, but +1 to not editing the `vendor` directory (so will need to revert that change). Made a suggestion below where I think it should go.

Reply:
I've rolled back the changes.